### PR TITLE
kubeadm: add kind based tests for older branches

### PR DESCRIPTION
**Pull request by neolit123**
- add kubeadm-kind-{1.14, 1.13, 1.12}
- add email notifications
- add presubmit branch jobs

/assign @krzyzacy @BenTheElder 
cc @kubernetes/sig-cluster-lifecycle-pr-reviews 

/kind feature
/priority important-longterm

### CODE REVIEW

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 1]
```diff
@@ -1,78 +1,243 @@
-presubmits:
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: moved the presubmit at the bottom as it's not going to change that often (same as master).

File: testgrid/config.yaml [hunk 2]
```diff
@@ -5811,9 +5818,32 @@ dashboards:
     alert_options:
       alert_mail_to_addresses: kubernetes-sig-cluster-lifecycle@googlegroups.com
     alert_stale_results_hours: 24
-    num_failures_to_alert: 4 # Runs every 6h. Alert when it's been failing for 24 hours
+    num_failures_to_alert: 6 # Runs every 2h. Alert when it's been failing for 12 hours
```
Comment (neolit123) [inline testgrid/config.yaml]: this was incorrect before - runs every 2 hours.
going to remove these k-a jobs completely eventually - once we resolve politics.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 3]
```diff
@@ -151,3 +316,79 @@ periodics:
       hostPath:
         path: /sys/fs/cgroup
         type: Directory
+
+# presubmit jobs
+
+presubmits:
+  kubernetes/kubernetes:
+  - name: pull-kubernetes-e2e-kind
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: this needs to be forked for old releases too

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: ok, will add the other branches too.

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: actually, is there an easy way to do that?
e.g. for a PR that is on `release-1.12` to call `/test pull-kuberentes-e2e-kind` to test this branch?

**Comment by neolit123** [comment]
please verify if i'm doing the presubmit / branches correctly.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 4]
```diff
@@ -151,3 +237,387 @@ periodics:
...
+      - name: modules
+        hostPath:
+          path: /lib/modules
+          type: Directory
+      - name: cgroup
+        hostPath:
+          path: /sys/fs/cgroup
+          type: Directory
+
+  kubernetes/kubernetes:
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: we don't need the repo key for each of these, you can have a list within the repo

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml [hunk 5]
```diff
@@ -151,3 +237,384 @@ periodics:
...
+      repo: kubeadm
+      base_ref: master
+      path_alias: k8s.io/kubeadm
+    - org: kubernetes-sigs
+      repo: kind
+      base_ref: master
+      path_alias: sigs.k8s.io/kind
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:latest-master
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kind.yaml]: we should use the `latest-1.14` etc. images for the branch specific jobs.

also we should get to stable images soon. current plan is to get kubetest2 to a usable state and then make a better kubetest2 + docker-in-docker image to use instead.

**Comment by neolit123** [comment]
updated to use latest-x branches for kubekins.

**Comment by BenTheElder** [comment]
thank you!

**Comment by neolit123** [comment]
looks like the kubekins for older branches (e.g. latest-1.12) has the kubetest with the kind deployer `--loglevel=debug` bug for `get` commands:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kind-1-12/2/build-log.txt
```
2019/02/24 00:47:35 kind.go:160: kind.go:getKubeConfigPath()
2019/02/24 00:47:35 process.go:153: Running: kind get kubeconfig-path --loglevel=debug --name=kind-kubetest
2019/02/24 00:47:35 process.go:155: Step 'kind get kubeconfig-path --loglevel=debug --name=kind-kubetest' finished in 117.468987ms
```

`--loglevel=debug` is not present in `:latest-master`.
if this is a case of tags/images not being pushed - no rush...

**Pull request merged**